ui/components/app/modals/export-private-key-modal/export-private-key-modal.component.js

```diff
@@ -14,101 +21,121 @@ import {
+  const [showWarning, setShowWarning] = useState(true);
+  const [showHoldToReveal, setShowHoldToReveal] = useState(false);
 
-  componentWillUnmount() {
-    this.props.clearAccountDetails();
-    this.props.hideWarning();
-  }
+  const trackEvent = useContext(MetaMetricsContext);
+  const t = useI18nContext();
 
-  exportAccountAndGetPrivateKey = (password, address) => {
-    const { exportAccount } = this.props;
+  useEffect(() => {
+    return () => {
+      clearAccountDetails();
+      hideWarning();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-    exportAccount(password, address)
-      .then((privateKey) => {
-        this.context.trackEvent({
+  const exportAccountAndGetPrivateKey = async (passwordInput, address) => {
+    try {
+      const privateKeyRetrieved = await exportAccount(passwordInput, address);
+      trackEvent(
+        {
           category: MetaMetricsEventCategory.Keys,
           event: MetaMetricsEventName.KeyExportRevealed,
           properties: {
             key_type: MetaMetricsEventKeyType.Pkey,
           },
-        });
-
-        this.setState({
-          privateKey,
-          showWarning: false,
-        });
-      })
-      .catch((e) => {
-        this.context.trackEvent({
+        },
+        {},
+      );
+      setPrivateKey(privateKeyRetrieved);
+      setShowWarning(false);
+      setShowHoldToReveal(true);
+    } catch (e) {
+      trackEvent(
+        {
           category: MetaMetricsEventCategory.Keys,
           event: MetaMetricsEventName.KeyExportFailed,
           properties: {
             key_type: MetaMetricsEventKeyType.Pkey,
             reason: 'incorrect_password',
           },
-        });
+        },
+        {},
+      );
 
-        log.error(e);
-      });
+      log.error(e);
+    }
   };
 
-  renderPasswordLabel(privateKey) {
+  const renderPasswordLabel = (privateKeyInput) => {
     return (
-      <span className="export-private-key-modal__password-label">
-        {privateKey
-          ? this.context.t('copyPrivateKey')
-          : this.context.t('typePassword')}
-      </span>
+      <Text
+        as="span"
+        color={Color.textDefault}
+        marginBottom={2}
+        variant={TextVariant.bodySm}
+      >
+        {privateKeyInput ? t('copyPrivateKey') : t('typePassword')}
+      </Text>
     );
-  }
+  };
 
-  renderPasswordInput(privateKey) {
-    const plainKey = privateKey && stripHexPrefix(privateKey);
+  const renderPasswordInput = (privateKeyInput) => {
+    const plainKey = privateKeyInput && stripHexPrefix(privateKeyInput);
 
-    if (!privateKey) {
+    if (!privateKeyInput) {
       return (
         <input
+          aria-label="input-password"
           type="password"
           className="export-private-key-modal__password-input"
-          onChange={(event) => this.setState({ password: event.target.value })}
+          onChange={(event) => setPassword(event.target.value)}
```

Comment:
Could use a `component-library` component here

Add these imports(You are already importing from these places so just add them may do it automatically when you type out the components and consts)

```
import { TextField, TEXT_FIELD_TYPES } from '../../../component-library';
```

```
<TextField
  type={TEXT_FIELD_TYPES.PASSWORD}
  className="export-private-key-modal__password-input"
  onChange={(event) => setPassword(event.target.value)}
/>
```

Then you can remove some css and just leave the styles for `__password-input` as
```
  &__password-input {
    width: 291px;
  }
```

Not a fan of random pixel values as width but because it's smaller than our smallest screen size I think it's fine

Reply:
sure, i'll change the to the TextField component